extensions/wikia/AdEngine/js/template/bfaaDesktop.js

```diff
@@ -72,24 +72,39 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 			recoveryTweaker.tweakSlot(params.slotName, iframe);
 		}
 
-		if (params.videoUrl && params.videoTriggerElement) {
-			slotTweaker.onReady(params.slotName, function() {
-				var divs = doc.querySelectorAll('#' + params.slotName + ' > div'),
-					imageContainer = divs[divs.length - 1],
-					video = uapVideoAd.init(doc.getElementById(params.slotName), imageContainer, params.videoUrl);
-
-				params.videoTriggerElement.addEventListener('click', function () {
-					uapVideoAd.playAndToggle(video, imageContainer);
-				});
+		if (videoEnabled(params)) {
+			videoAd.onLibraryReady(function () {
+				var playTrigger = videoAd.setupVideo(
```

Comment:
For me it was a bit confusing at the beginning. What do you think about renaming videoAd module to videoAdFactory with create method, and then create method which returns object with interface:
```
{
  play: ...
}
```